code/obj/flock/cage.dm

```diff
@@ -1,41 +1,60 @@
 /////////////////////////////////////////////////////////////////////////////////
 // ENERGY CAGE
 /////////////////////////////////////////////////////////////////////////////////
-// it's just an ice cube, but stronger and it looks different
-// and eats people, i guess, too
-/obj/icecube/flockdrone
+/obj/flock_structure/cage
 	name = "weird energy cage"
 	desc = "You can see the person inside being rapidly taken apart by fibrous mechanisms. You ought to do something about that."
 	icon = 'icons/misc/featherzone.dmi'
 	icon_state = "cage"
-	steam_on_death = 0
+	flock_id = "matter reprocessor"
+	flags = USEDELAY
 	health = 30
+	health_max = 30
 	alpha = 192
-	var/datum/flock/flock
-	var/mob/living/occupant = null // todo: make this work with more than just humans (borgs, critters, probably not cubes)
+	anchored = FALSE
+	var/atom/occupant = null
 	var/obj/target = null
-	var/eating_occupant = 0
+	var/eating_occupant = FALSE
 	var/initial_volume = 200
 	// convert things into different fluids, convert those fluids into coagulated gnesis, convert 50 of that into an egg
 	var/target_fluid = "flockdrone_fluid"
 	var/create_egg_at_fluid = 100
 	var/absorb_per_process_tick = 2
-	mat_changename = 0
-	mat_changedesc = 0
-	mat_changeappearance = 0
+	mat_changeappearance = FALSE
 
 
-	New(loc, mob/living/iced as mob, datum/flock/F=null)
-		..()
-		src.flock = F
+	New(loc, var/atom/iced, datum/flock/F=null)
+		..(loc,F)
+		if(iced && !isAI(iced) && !isblob(iced) && !iswraith(iced))
+			if(istype(iced.loc, /obj/flock_structure/cage))
+				qdel(src)
+				return
+
+			if(!(ismob(iced) || iscritter(iced)))
+				qdel(src)
+				return
+			iced:set_loc(src)
+
+			boutput(iced, "<span class='alert'>You are trapped within [src]!</span>")
+
 		var/datum/reagents/R = new /datum/reagents(initial_volume)
 		src.reagents = R
-		R.my_atom = src //grumble
+		R.my_atom = src
 		if(iced)
```

Comment:
should use `src.create_reagents(initial_volume)`

Reply:
https://github.com/stonepillars/goon-flock/pull/333